src/library_browser.js

```diff
@@ -1268,7 +1268,7 @@ mergeInto(LibraryManager.library, {
       }
       var data = msg.data['data'];
       if (data) {
-        if (!data.byteLength) data = new Uint8Array(data);
+        data = new Uint8Array(data);
```

Comment:
It's not a typed array (or at least it's not meant to be!). In this PR only ArrayBuffers are sent as the `data` property, and `new Uint8Array(<ArrayBuffer>)` doesn't copy the array, it creates a view on the buffer, at least according to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray#Syntax